pkg/app/rds_postgres.go

```diff
@@ -255,63 +266,74 @@ func (pdb *RDSPostgresDB) Ping(ctx context.Context) error {
 		return errors.New("Databases are missing from configmap")
 	}
 
-	var connectionString string = fmt.Sprintf("host=%s user=%s password=%s dbname=%s sslmode=disable", dbconfig.Data["postgres.host"], dbconfig.Data["postgres.user"], dbsecret.Data["password"], databases[0])
+	isReadyCommand := fmt.Sprintf(postgresConnectionString+"'SELECT version();'", dbsecret.Data["password"], dbconfig.Data["postgres.host"], dbconfig.Data["postgres.user"], databases[0])
 
-	// Initialize connection object.
-	db, err := sql.Open("postgres", connectionString)
-	if err != nil {
-		return err
-	}
+	pingCommand := []string{"sh", "-c", isReadyCommand}
 
-	err = db.Ping()
+	_, stderr, err := pdb.execCommand(ctx, pingCommand)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "Error while Pinging the database: %s, app: %s", stderr, pdb.name)
 	}
-
-	pdb.sqlDB = db
-	log.Info().Print("Connected to database.", field.M{"app": pdb.name})
+	log.Print("Ping to the application was successful.", field.M{"app": pdb.name})
 	return nil
 }
 
 func (pdb RDSPostgresDB) Insert(ctx context.Context) error {
+	log.Print("Adding entry to database", field.M{"app": pdb.name})
 	now := time.Now().Format(time.RFC3339Nano)
-	stmt := "INSERT INTO inventory (name) VALUES ($1);"
-	_, err := pdb.sqlDB.Exec(stmt, now)
+	insert := fmt.Sprintf(postgresConnectionString+
+		"\"INSERT INTO inventory (name) VALUES ('%s');\"", pdb.password, pdb.host, pdb.username, pdb.databases[0], now)
+
+	insertQuery := []string{"sh", "-c", insert}
+	_, stderr, err := pdb.execCommand(ctx, insertQuery)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "Error while inserting data into table: %s, app: %s", stderr, pdb.name)
 	}
 	log.Info().Print("Inserted a row in test db.", field.M{"app": pdb.name})
 	return nil
 }
 
 func (pdb RDSPostgresDB) Count(ctx context.Context) (int, error) {
-	stmt := "SELECT COUNT(*) FROM inventory;"
-	row := pdb.sqlDB.QueryRow(stmt)
-	var count int
-	err := row.Scan(&count)
+	log.Print("Counting entries from database", field.M{"app": pdb.name})
+	count := fmt.Sprintf(postgresConnectionString+
+		"\"SELECT COUNT(*) FROM inventory;\"", pdb.password, pdb.host, pdb.username, pdb.databases[0])
+
+	countQuery := []string{"sh", "-c", count}
+	stdout, stderr, err := pdb.execCommand(ctx, countQuery)
```

Comment:
```suggestion
	countQuery := fmt.Sprintf(postgresConnectionString+
		"\"SELECT COUNT(*) FROM inventory;\"", pdb.password, pdb.host, pdb.username, pdb.databases[0])

	countCommand := []string{"sh", "-c", count}
	stdout, stderr, err := pdb.execCommand(ctx, countQuery)
```

Let's do this at other places as well, if possible.